ipywidgets/widgets/widget_selection.py

```diff
@@ -161,7 +168,8 @@ def __init__(self, *args, **kwargs):
         # We have to make the basic options bookkeeping consistent
         # so we don't have errors the first time validators run
         self._initializing_traits_ = True
-        kwargs['options'] = self._validate_options(Bunch(value = kwargs.get('options', ())))
+        kwargs['options'] = _exhaust_iterable(kwargs.get('options', ()))
+        self._options_full = _make_options(kwargs['options'])
```

Comment:
My point here was that instead of repeating the logic of assigning self._options_full, we should consolidate the logic and call _validate_options when we needed. I think I would rather have _validate_options access `proposal` as a dict, and keep `_options_full` only being assigned there. Thoughts?

Reply:
My main concern was that someone might be overriding `_validate_options` due to this pattern being enforced previously, and in the traitlets docs it says:

```
The proposal dictionary must hold the following keys:

    * ``owner`` : the HasTraits instance
    * ``value`` : the proposed value for the modified trait attribute
    * ``trait`` : the TraitType instance associated with the attribute
```

The `must` is what caused me to change it. We can probably add another method that gets called by both init and the validators though? That would also fix us currently repeating the logic in the subclasses' validators.

Reply:
> We can probably add another method that gets called by both init and the validators though? That would also fix us currently repeating the logic in the subclasses' validators.

That sounds like a better way. +1.

Reply:
On second thought, I like the way you did things. I'll merge this PR as-is. Thanks again!